pyart/retrieve/simple_moment_calculations.py

```diff
@@ -66,3 +70,200 @@ def calculate_snr_from_reflectivity(
     snr_dict = get_metadata(snr_field)
     snr_dict['data'] = pseudo_power - noise_floor_estimate
     return snr_dict
+
+
+def compute_noisedBZ(nrays, noisedBZ_val, range, ref_dist,
+                     noise_field=None):
+    """
+    Computes noise in dBZ from reference noise value.
+
+    Parameters
+    ----------
+    nrays: int
+        number of rays in the reflectivity field
+
+    noisedBZ_val: float
+        Estimated noise value in dBZ at reference distance
+
+    range: np array of floats
+        range vector in m
+
+    ref_dist: float
+        reference distance in Km
+
+    noise_field: str
+        name of the noise field to use
+
+    Returns
+    -------
+    noisedBZ : dict
+        the noise field
+
+    """
+    # parse the field parameters
+    if noise_field is None:
+        noise_field = get_field_name('noisedBZ_hh')
+
+    noisedBZ_vec = noisedBZ_val+20.*np.ma.log10(1e-3*range/ref_dist)
+
+    noisedBZ = get_metadata(noise_field)
+    noisedBZ['data'] = np.tile(noisedBZ_vec, (nrays, 1))
+    noisedBZ['data'].set_fill_value(get_fillvalue())
+
+    return noisedBZ
+
+
+def compute_snr(radar, refl_field=None, noise_field=None, snr_field=None):
+    """
+    Computes SNR from a reflectivity field and the noise in dBZ.
+
+    Parameters
+    ----------
+    radar : Radar
+        radar object
+
+    refl_field, noise_field : str
+        name of the reflectivity and noise field used for the calculations
+
+    snr_field : str
+        name of the SNR field
+
+    Returns
+    -------
+    snr : dict
+        the SNR field
+
+    """
+    # parse the field parameters
+    if refl_field is None:
+        refl_field = get_field_name('reflectivity')
+    if noise_field is None:
+        noise_field = get_field_name('noisedBZ_hh')
+    if snr_field is None:
+        snr_field = get_field_name('signal_to_noise_ratio')
+
+    # extract fields from radar
+    if refl_field in radar.fields:
+        refl = radar.fields[refl_field]['data']
+    else:
+        raise KeyError('Field not available: ' + refl_field)
+    if noise_field in radar.fields:
+        noisedBZ = radar.fields[noise_field]['data']
+    else:
+        raise KeyError('Field not available: ' + noise_field)
+
+    mask = np.ma.getmaskarray(refl)
+    fill_value = refl.get_fill_value()
+
+    snr_data = refl-noisedBZ
+    snr_data.set_fill_value(fill_value)
+    snr_data.data[mask.nonzero()] = fill_value
```

Comment:
`snr_data[mask] = np.ma.masked` although this should have propagated from the refl array automatically.